service/start.go

```diff
@@ -119,6 +115,132 @@ func (svc *service) startNostr(ctx context.Context, encryptionKey string) error
 	return nil
 }
 
+type createAppSubscriber struct {
+	events.EventSubscriber
+	svc   *service
+	relay *nostr.Relay
+}
+
+func (s *createAppSubscriber) ConsumeEvent(ctx context.Context, event *events.Event, globalProperties map[string]interface{}) {
+	switch event.Event {
+	case "app_created":
+		properties, ok := event.Properties.(map[string]interface{})
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to cast event.Properties to map")
+			return
+		}
+		walletChildPubkey, ok := properties["walletChildPubkey"].(string)
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to get app walletChildPubkey")
+			return
+		}
+		id, ok := properties["id"].(uint)
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to get app id")
+			return
+		}
+		if walletChildPubkey != "" {
+			go func() {
+				appWalletPrivKey, err := s.svc.keys.GetBIP32ChildKey(uint32(id))
+				if err != nil {
+					logger.Logger.WithError(err).Error("Failed to calculate app wallet priv key")
+				}
+				err = s.svc.startAppWalletSubscription(ctx, s.relay, walletChildPubkey, appWalletPrivKey)
+				if err != nil {
+					logger.Logger.WithError(err).Error("Failed to subscribe to wallet")
+				}
+				// TODO what should we do?
+			}()
+		}
+	}
+}
+
+type deleteAppSubscriber struct {
+	events.EventSubscriber
+	walletChildPubkey string
+	relay             *nostr.Relay
+	nostrSubscription *nostr.Subscription
+	svc               *service
+	infoEventId       string
+}
+
+func (s *deleteAppSubscriber) ConsumeEvent(ctx context.Context, event *events.Event, globalProperties map[string]interface{}) {
+	switch event.Event {
+	case "app_deleted":
+		properties, ok := event.Properties.(map[string]interface{})
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to cast event.Properties to map")
+			return
+		}
+
+		walletChildPubkey, ok := properties["walletChildPubkey"].(string)
+		if s.walletChildPubkey == walletChildPubkey {
+			id, _ := properties["id"].(uint)
+			s.nostrSubscription.Unsub()
+			appWalletPrivKey, _ := s.svc.keys.GetBIP32ChildKey(uint32(id))
+			err := s.svc.nip47Service.PublishNip47InfoDeletion(ctx, s.relay, walletChildPubkey, appWalletPrivKey, s.infoEventId)
+			if err != nil {
+				logger.Logger.WithField("event", event).Error("Failed to publish nip47 info deletion")
+			}
+		}
+	}
+}
+
+func (svc *service) startAllExistingAppsWalletSubscriptions(ctx context.Context, relay *nostr.Relay) {
+	var apps []db.App
+	result := svc.db.Where("wallet_child_pubkey != ?", "").Find(&apps)
+	if result.Error != nil {
+		logger.Logger.WithError(result.Error).Error("Failed to fetch App records with non-empty WalletChildPubkey")
+		return
+	}
+
+	for _, app := range apps {
+		go func(app db.App) {
+			if app.WalletChildPubkey != "" {
+				err := svc.startAppWalletSubscription(ctx, relay, app.WalletChildPubkey, "")
+				if err != nil {
+					logger.Logger.WithError(err).Error("Failed to subscribe to wallet")
```

Comment:
Can we use `.WithFields` and include the app ID or something to identify what wallet the error was for?